apps/frontend/src/app/5_pages/AavePage/components/LendPositionsList/components/ToggleCollateralAction/ToggleCollateralAction.tsx

```diff
@@ -10,10 +11,21 @@ type ToggleCollateralActionProps = {
 
 export const ToggleCollateralAction: FC<ToggleCollateralActionProps> = ({
   position,
-}) => (
-  <Toggle
-    className="[&_*]:ml-0"
-    checked={position.collateral}
-    onChange={() => 'TODO: implement'}
-  />
-);
+}) => {
+  const { handleSwitchCollateral } = useAaveSupply();
+  const [isCollateral, setIsCollateral] = useState(position.collateral);
+
+  const toggleCollateral = useCallback(() => {
+    handleSwitchCollateral(position.asset, !isCollateral, {
+      onComplete: () => setIsCollateral(!isCollateral),
+    });
+  }, [handleSwitchCollateral, setIsCollateral, isCollateral, position.asset]);
```

Comment:
should be better await and then call the onComplete here? I prefer not changing the states outside of the component